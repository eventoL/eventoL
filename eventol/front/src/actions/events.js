export const SET_EVENTS = 'SET_EVENTS';
export const UPDATE_EVENT = 'UPDATE_EVENT';
export const DELETE_EVENT = 'DELETE_EVENT';
export const CREATE_EVENT = 'CREATE_EVENT';

export const setEvents = events => ({type: SET_EVENTS, events});
export const updateEvent = event => ({type: UPDATE_EVENT, event});
export const deleteEvent = event => ({type: DELETE_EVENT, event});
export const createEvent = event => ({type: CREATE_EVENT, event});

const eventsActions = {
  setEvents,
  createEvent,
  updateEvent,
  deleteEvent,
  DELETE_EVENT,
  CREATE_EVENT,
  UPDATE_EVENT,
  SET_EVENTS,
};
export default eventsActions;
