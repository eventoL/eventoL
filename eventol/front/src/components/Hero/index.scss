@import '../../styles/index';

.hero {
  background-size: cover;
  margin-bottom: -22vh;
  min-height: 800px;
  position: relative;
  width: 100%;

  .btn-create-event {
    line-height: 1.4;
    font-weight: bold;
    padding: 10px 15px;
    border-radius: 15px;
  }

  .content {
    -ms-flex: 1;
    -webkit-box-flex: 1;
    box-sizing: border-box;
    flex: 1;
    position: relative;
    text-align: center;
    top: 20vh;
    z-index: 5;

    .logo {
      z-index: 4;
      max-width: 250px;
      max-height: 50vh;
    }

    h2 {
      color: $text-color;
      font-size: 27px;
      font-weight: unset;
    }

    p {
      color: $text-color;
      width: 100%;
    }

    .button-wrapper {
      @extend %flex-element;
      margin-top: 30px;
      width: 400px;

      .button:first-child {
        margin-right: 10px;
      }
    }
  }

  .overlay {
    -webkit-transform: rotate(180deg);

    // FF3.6-15
    background: -webkit-linear-gradient(top, $gradient-gray-color 20%, $gradient-end-rgba 100%);
    // Chrome10-25, Safari5.1-6
    background: linear-gradient(to bottom, $gradient-gray-color 20%, $gradient-end-rgba 100%);
    // W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#a6000000', endColorstr='#00000000',GradientType=0 );

    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    transform: rotate(180deg);
    width: 100%;
    z-index: 3;
  }

}
