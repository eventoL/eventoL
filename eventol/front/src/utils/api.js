import _ from 'lodash';
import Cookies from 'js-cookie';

import Logger from './logger';
import {getReportUrl} from './urls';

export const getCsrf = () => Cookies.get('csrftoken');

export const addQueryString = (url, query) => {
  const queryKeys = query ? Object.keys(query) : [];
  if (!_.isEmpty(queryKeys)) {
    const params = queryKeys.map(k => `${k}=${encodeURIComponent(query[k])}`);
    return `${url}?${params.join('&')}`;
  }
  return url;
};

export const genericFetch = (url, queryString, params) => {
  const newUrl = addQueryString(url, queryString);
  return fetch(newUrl, params)
    .then(res => {
      if (res.ok) {
        return res.json();
      }
      return Promise.reject(res);
    })
    .catch(res => {
      if (res.status === 403) {
        window.location.hash = '/';
        window.location.href = '/';
        Logger.warning('Redirect to /', res.status, res.statusText);
      } else {
        Logger.error('There has been an error', res.status, res.statusText);
      }
    });
};

export const getUrl = (url, queryString) =>
  genericFetch(url, queryString, {
    method: 'GET',
    credentials: 'same-origin',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      'X-CSRFToken': getCsrf(),
    },
  });

export const postUrl = (url, data, queryString) =>
  genericFetch(url, queryString, {
    method: 'POST',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'X-CSRFToken': getCsrf(),
    },
    body: JSON.stringify(data),
    credentials: 'same-origin',
  });

export const loadReports = (pageSize, page, sorted) =>
  getUrl(getReportUrl(pageSize, page, sorted));
