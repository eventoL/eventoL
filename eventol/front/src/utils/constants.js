export const MOBILE_WIDTH = 950;
export const LOGO_LANDING_DEFAULT = '/static/manager/img/logo.png';
export const BACKGROUND_DEFAULT = '/static/manager/img/background.png';
export const LOGO_HEADER_DEFAULT = '/static/manager/img/eventol-white.png';
export const REPORT_REQUIRED_FIELDS =
  'name,event_slug,email,location,report,id,tags';
export const HOME_REQUIRED_FIELDS =
  'event_slug,place,image,name,attendees_count,abstract,tags';
export const MAP_LAYER =
  'https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png';

export const SLIDER_BASE_SETTINGS = {
  variableWidth: false,
  dots: true,
  infinite: false,
  speed: 500,
  slidesToShow: 4,
  slidesToScroll: 4,
  initialSlide: 0,
  responsive: [
    {
      breakpoint: 1290,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 3,
      },
    },
    {
      breakpoint: 900,
      settings: {
        slidesToShow: 2,
        slidesToScroll: 2,
        initialSlide: 2,
      },
    },
    {
      breakpoint: 610,
      settings: {
        slidesToShow: 1,
        slidesToScroll: 1,
      },
    },
  ],
};

export const MAP_SETTINGS = {
  scrollWheelZoom: false,
  zoomControl: false,
  dragging: false,
  keyboard: false,
  doubleClickZoom: false,
  touchZoom: false,
  boxZoom: false,
  minZoom: 11,
  maxZoom: 11,
};
