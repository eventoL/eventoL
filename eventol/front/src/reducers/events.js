import _ from 'lodash';
import {
  SET_EVENTS, UPDATE_EVENT,
  DELETE_EVENT, CREATE_EVENT,
} from '../actions/events';
import {getEvent} from '../utils/events';

const initState = {
  events: [],
};

const updateEvent = (state, {pk, data}) => {
  const events = [...state.events];
  const index = _.findIndex(events, {pk});
  events[index] = getEvent(pk, data);
  return {...state, events};
};

const createEvent = (state, {pk, data}) => (
  {...state, events: [...state.events, getEvent(pk, data)]}
);

const deleteEvent = (state, {pk}) => {
  const events = [...state.events];
  _.remove(events, event => event.pk === pk);
  return {...state, events};
};

const setEvents = (state, events) => ({...state, events});

export default (state = initState, action) => {
  switch (action.type){
    case SET_EVENTS: return setEvents(state, action.events);
    case UPDATE_EVENT: return updateEvent(state, action.event);
    case CREATE_EVENT: return createEvent(state, action.event);
    case DELETE_EVENT: return deleteEvent(state, action.event);
    default: return state;
  }
};
