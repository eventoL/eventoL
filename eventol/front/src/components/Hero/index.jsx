import React from 'react';
import PropTypes from 'prop-types';

import {CREATE_EVENT} from '../../utils/urls';
import {getSlugParsed} from '../../utils/events';
import {BACKGROUND_DEFAULT, LOGO_LANDING_DEFAULT} from '../../utils/constants';

import './index.scss';

export default class Hero extends React.PureComponent {
  static propTypes = {
    background: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ]),
    isLogged: PropTypes.bool,
    logoLanding: PropTypes.string,
    message: PropTypes.string,
    slug: PropTypes.string,
  };

  static defaultProps = {
    background: BACKGROUND_DEFAULT,
    children: null,
    isLogged: false,
    logoLanding: LOGO_LANDING_DEFAULT,
    message: null,
    slug: null,
  };

  getMessage() {
    const {slug, message} = this.props;
    if (message) {
      return <h2>{message}</h2>;
    }
    if (slug) {
      const parsedSlug = getSlugParsed(slug);
      return (
        <h2>
          {`${gettext('You are seeing all of')} ${parsedSlug} ${gettext(
            'events'
          )}`}
          <br />
          {`${gettext('Please, select one to continue')}`}
        </h2>
      );
    }
    return <h2>{gettext('Search your event')}</h2>;
  }

  render() {
    const {children, background, logoLanding, isLogged} = this.props;
    const backgroundImage = `url(${background})`;
    return (
      <div className="hero" id="hero" style={{backgroundImage}}>
        <div className="content">
          <p>
            <img alt="logo" className="logo" src={logoLanding} />
          </p>
          {isLogged && (
            <a
              className="btn btn-raised btn-warning btn-create-event"
              href={CREATE_EVENT}
            >
              {gettext('Create event')}
            </a>
          )}
          {this.getMessage()}
          {children}
        </div>
        <div className="overlay" />
      </div>
    );
  }
}
