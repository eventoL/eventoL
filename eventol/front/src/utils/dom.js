import {MOBILE_WIDTH} from './constants';

export const focusOn = id => {
  document.getElementById(id).focus();
};

export const mapSizesToProps = ({width}) => ({
  isMobile: width < MOBILE_WIDTH,
});
