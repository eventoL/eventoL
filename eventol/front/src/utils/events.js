import {getEventUrl} from './urls';
import {LOGO_LANDING_DEFAULT} from './constants';


export const getEvent = (pk, fields) => ({pk, fields});
export const getSlugParsed = slug => slug
  .toLowerCase()
  .split('-')
  .join(' ')
  .split('_')
  .join(' ')
  .split(' ')
  .map(word => ((word) ? word.replace(word[0], word[0].toUpperCase()) : ''))
  .join(' ');

export const parseEventToItem = ({
  tags, place, image,
  name: title, attendees_count: attendees,
  abstract: overview, event_slug: eventSlug,
}) => {
  let backdrop = image;
  if (image){
    backdrop = new URL(image).pathname;
  }
  return {
    eventSlug,
    title,
    attendees,
    overview,
    backdrop,
    place,
    tags,
    key: eventSlug,
    url: getEventUrl(eventSlug),
  };
};

export const emptyEventItem = {
  key: 'not_found',
  title: gettext('Event not found'),
  overview: gettext('No Event found in your search'),
  backdrop: LOGO_LANDING_DEFAULT,
};
