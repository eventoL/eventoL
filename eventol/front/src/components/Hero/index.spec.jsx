import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../utils/events', () => ({
  getSlugParsed: jest.fn().mockImplementation(() => 'parsed-slug'),
}));

jest.mock('../../utils/constants', () => ({
  BACKGROUND_DEFAULT: 'BACKGROUND_DEFAULT',
  LOGO_LANDING_DEFAULT: 'LOGO_LANDING_DEFAULT',
}));

jest.mock('../../utils/urls', () => ({
  CREATE_EVENT: 'CREATE_EVENT',
}));

import {getSlugParsed} from '../../utils/events';

import Hero from '.';

describe('Hero', () => {
  let component;
  let tree;
  let background;
  let logoLanding;
  let message;
  let slug;
  let isLogged;

  const renderHero = () => {
    component = renderer.create(
      <Hero
        background={background}
        isLogged={isLogged}
        logoLanding={logoLanding}
        message={message}
        slug={slug}
      />
    );
    tree = component.toJSON();
  };

  beforeEach(() => {
    background = undefined;
    logoLanding = undefined;
    message = undefined;
    slug = undefined;
    isLogged = undefined;
  });

  test('Default render', () => {
    renderHero();
    expect(tree).toMatchSnapshot();
  });

  test('With message', () => {
    message = 'message';
    renderHero();
    expect(tree).toMatchSnapshot();
  });

  test('With slug', () => {
    slug = 'slug';
    renderHero();

    expect(tree).toMatchSnapshot();
    expect(getSlugParsed).toBeCalled();
    expect(getSlugParsed).toBeCalledWith(slug);
  });

  test('With custom backgound', () => {
    background = 'custom-background';
    renderHero();
    expect(tree).toMatchSnapshot();
  });

  test('With custom logo', () => {
    logoLanding = 'logo-custom';
    renderHero();
    expect(tree).toMatchSnapshot();
  });

  test('With user (isLogged is true)', () => {
    isLogged = true;
    renderHero();
    expect(tree).toMatchSnapshot();
  });
});
