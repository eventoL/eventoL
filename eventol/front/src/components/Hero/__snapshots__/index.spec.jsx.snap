// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Default render 1`] = `
<div
  className="hero"
  id="hero"
  style={
    Object {
      "backgroundImage": "url(BACKGROUND_DEFAULT)",
    }
  }
>
  <div
    className="content"
  >
    <p>
      <img
        alt="logo"
        className="logo"
        src="LOGO_LANDING_DEFAULT"
      />
    </p>
    <h2>
      Search your event
    </h2>
  </div>
  <div
    className="overlay"
  />
</div>
`;

exports[`Hero With custom backgound 1`] = `
<div
  className="hero"
  id="hero"
  style={
    Object {
      "backgroundImage": "url(custom-background)",
    }
  }
>
  <div
    className="content"
  >
    <p>
      <img
        alt="logo"
        className="logo"
        src="LOGO_LANDING_DEFAULT"
      />
    </p>
    <h2>
      Search your event
    </h2>
  </div>
  <div
    className="overlay"
  />
</div>
`;

exports[`Hero With custom logo 1`] = `
<div
  className="hero"
  id="hero"
  style={
    Object {
      "backgroundImage": "url(BACKGROUND_DEFAULT)",
    }
  }
>
  <div
    className="content"
  >
    <p>
      <img
        alt="logo"
        className="logo"
        src="logo-custom"
      />
    </p>
    <h2>
      Search your event
    </h2>
  </div>
  <div
    className="overlay"
  />
</div>
`;

exports[`Hero With message 1`] = `
<div
  className="hero"
  id="hero"
  style={
    Object {
      "backgroundImage": "url(BACKGROUND_DEFAULT)",
    }
  }
>
  <div
    className="content"
  >
    <p>
      <img
        alt="logo"
        className="logo"
        src="LOGO_LANDING_DEFAULT"
      />
    </p>
    <h2>
      message
    </h2>
  </div>
  <div
    className="overlay"
  />
</div>
`;

exports[`Hero With slug 1`] = `
<div
  className="hero"
  id="hero"
  style={
    Object {
      "backgroundImage": "url(BACKGROUND_DEFAULT)",
    }
  }
>
  <div
    className="content"
  >
    <p>
      <img
        alt="logo"
        className="logo"
        src="LOGO_LANDING_DEFAULT"
      />
    </p>
    <h2>
      You are seeing all of parsed-slug events
      <br />
      Please, select one to continue
    </h2>
  </div>
  <div
    className="overlay"
  />
</div>
`;

exports[`Hero With user (isLogged is true) 1`] = `
<div
  className="hero"
  id="hero"
  style={
    Object {
      "backgroundImage": "url(BACKGROUND_DEFAULT)",
    }
  }
>
  <div
    className="content"
  >
    <p>
      <img
        alt="logo"
        className="logo"
        src="LOGO_LANDING_DEFAULT"
      />
    </p>
    <a
      className="btn btn-raised btn-warning btn-create-event"
      href="CREATE_EVENT"
    >
      Create event
    </a>
    <h2>
      Search your event
    </h2>
  </div>
  <div
    className="overlay"
  />
</div>
`;
